src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -2911,11 +2946,48 @@ def mi_ad_admin_delete(
     '''
     Deletes a managed instance active directory administrator.
     '''
+
     return client.delete(
         resource_group_name=resource_group_name,
         managed_instance_name=managed_instance_name
     )
 
+
+#####
+#           sql managed instance aad-only
+#####
+
+
+def mi_aad_only_disable(
+        client,
+        resource_group_name,
+        managed_instance_name):
+    '''
+    Disables the managed instance AAD-only setting
+    '''
+
+    return client.create_or_update(
```

Comment:
If it is long running operation and you want to support no wait, besides adding support_no_wait_true, you also need to wrap the operation like
https://github.com/Azure/azure-cli/blob/1ef59bace4dfad9cba137f49c8974aca4b0fff90/src/azure-cli/azure/cli/command_modules/sql/custom.py#L813-L818